test/test_distributed.py

```diff
@@ -778,6 +778,18 @@ def forward(self, x):
                 x = self.fc3(x)
                 return F.softmax(x, dim=1)
 
+        def model_step(model):
+            for layer in model.modules():
+                if isinstance(layer, nn.Linear):
+                    layer.weight.data = layer.weight.data + layer.weight.grad
+                    layer.weight.grad = None
+
+        def assert_equal_model(model_gpu, model_DDP):
+            for layer_gpu, layer_DDP in zip(model_gpu.modules(), model_DDP.module.modules()):
```

Comment:
assert their length is equal before you do `zip` - it terminates once *any* of the sequences terminate, so we might skip some checks

Reply:
Also, can't you just iterate over `.parameters()`? This will give you bias checks for free as well (we're not doing them)